src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public OSName(string name)
+        {
+            _name = name;
```

Comment:
Just because `default(OSName)` can be used to create an `OSName` with a null `_name` doesn't mean you shouldn't do parameter validation for the constructor. Why would anyone want to do `new OSName(null)` or `new OSName("")`?